examples/examples/tower-load.rs

```diff
@@ -1,410 +1,388 @@
+        use std::str;
+        let body = str::from_utf8(&bytes.as_ref()).map_err(|e| format!("{}", e))?;
+        trace!(request.body = ?body);
+        let new_filter = body
+            .parse::<tracing_subscriber::filter::EnvFilter>()
+            .map_err(|e| format!("{}", e))?;
+        self.handle.reload(new_filter).map_err(|e| format!("{}", e))
+    }
+}
+
+fn rsp(status: StatusCode, body: impl Into<Body>) -> Response<Body> {
+    Response::builder()
+        .status(status)
+        .body(body.into())
+        .expect("builder with known status code must not fail")
+}
+
+impl HandleError {
+    fn bad_request(e: impl std::error::Error + Send + 'static) -> Self {
+        HandleError::BadRequest(Box::new(e))
+    }
+}
+
+impl fmt::Display for HandleError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            HandleError::BadPath => f.pad("path must be a single ASCII character"),
+            HandleError::NoContentLength => f.pad("request must have Content-Length header"),
+            HandleError::BadRequest(ref e) => write!(f, "bad request: {}", e),
+            HandleError::Unknown => f.pad("unknown internal error"),
+        }
+    }
+}
+
+impl std::error::Error for HandleError {}
+
+impl fmt::Display for WrongMethod {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "unsupported method: please use one of {:?}", self.0)
+    }
+}
+
+impl std::error::Error for WrongMethod {}
+
+fn gen_uri(authority: &str) -> (usize, String) {
+    static ALPHABET: &str = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    let mut rng = rand::thread_rng();
+    let idx = rng.gen_range(0, ALPHABET.len() + 1);
+    let len = rng.gen_range(0, 26);
+    let letter = ALPHABET.get(idx..=idx).unwrap_or("");
+    (len, format!("http://{}/{}", authority, letter))
+}
+
+#[tracing::instrument(target = "gen", "load_gen")]
+async fn load_gen(addr: SocketAddr) -> Result<(), Err> {
+    let svc = ServiceBuilder::new()
+        .buffer(5)
+        .layer(request_span::layer(req_span))
+        .timeout(Duration::from_millis(200))
+        .service(Client::new());
+
+    while let Some(_) = time::interval(Duration::from_millis(50)).next().await {
+        let authority = format!("{}", addr);
+        let mut svc = svc.clone();
+        svc.ready().await?;
+
+        let f = async move {
+            let sleep = rand::thread_rng().gen_range(0, 25);
+            time::delay_for(Duration::from_millis(sleep)).await;
+
+            let (len, uri) = gen_uri(&authority);
+            let req = Request::get(&uri[..])
+                .header("Content-Length", len)
+                .body(Body::empty())
+                .unwrap();
+
+            let span = tracing::debug_span!(
+                target: "gen",
+                "request",
+                req.method = ?req.method(),
+                req.path = ?req.uri().path(),
+            );
+            let _s = span.enter();
```

Comment:
I think this `span.enter()` call is gonna hit the issue we've outlined here: https://github.com/tokio-rs/tracing#in-asynchronous-code. Any thoughts how we could structure this code in an idiomatic way, @hawkw?

Reply:
Yeah, that's correct --- this used to be in a future `poll` implementation, IIRC. I think we should either stick the rest of the function into an async block, like:
```rust
            let span = tracing::debug_span!(
                target: "gen",
                "request",
                req.method = ?req.method(),
                req.path = ?req.uri().path(),
            );
            async move {
                // ... rest of the code goes here ...
            }.instrument(span)
```
or consider factoring out some of this code into a separate `async fn` (not sure if that makes sense, since we close over a lot of state)?

Reply:
I've opted for nesting another span because we _do_ close over a lot of state.